### PR TITLE
Pressure valves

### PR DESCRIPTION
**Pull request by jellyveggie2**
Adds manual and digital pressure valves.
https://files.catbox.moe/fptnu0.webm

They open and close depending on a side input's pressure and their settings, so you could set a release valve to open up if pressure in a pipeline on the SME reaches 1000 kPa, or set one of those emergency air tanks in maint to release their load if distro ever goes below 90kPa:

Emergency supply

On IRL pressure valves, there's just a piston pitted against a spring you regulate to be weaker or stronger to require a different threshold pressure, so both the manual and digital valves require no power to function. Digital valves will still require some power for you to be able to modify their settings however.
Can be controled by AAC

:cl:
 * rscadd: Added manual and digital pressure valves. If powered on, they'll automatically open and close depending on a side input's pressure, which must be above the configured pressure (or below, if you've configured it that way).

### CODE REVIEW

**Comment by SonixApache** [comment]
what does this do that normal digital valves controlled through the AAC don't?

**Comment by WindowsErrors** [comment]
use less brain power

File: code/ATMOSPHERICS/components/trinary_devices/pressure_valve.dm [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+// NPCs and assorted gremlins
+/obj/machinery/atmospherics/trinary/pressure_valve/npc_tamper_act(mob/living/L)
+	if (prob(33))
+		open_on_above_threshold = !open_on_above_threshold
+		investigation_log(I_ATMOS,"was switched to open [(open_on_above_threshold ? "above" : "below")] [pressure_threshold]kPa by [key_name(L)]")
+
+	else if (prob(33))
+		pressure_threshold = rand(0, 9000)
+		investigation_log(I_ATMOS,"had it's threshold configured to open [(open_on_above_threshold ? "above" : "below")] [pressure_threshold]kPa by [key_name(L)]")
```
Comment (Exxion) [inline code/ATMOSPHERICS/components/trinary_devices/pressure_valve.dm]:
```suggestion
		investigation_log(I_ATMOS,"had its threshold configured to open [(open_on_above_threshold ? "above" : "below")] [pressure_threshold]kPa by [key_name(L)]")
```

**Comment by jellyveggie2** [comment]
> what does this do that normal digital valves controlled through the AAC don't?

They'll work even without power, and you won't necessarily have to run around linking valves everywhere to an AAC somewhere. Other than that, not much.

**Comment by WindowsErrors** [comment]
i usually just eyeball a gas mix anyways

**Comment by SonixApache** [comment]
seems absolutely redundant to me but i'm not an atmosmain so whatever i guess

**Comment by Cyadox** [comment]
I would argue that this oversimplifies the usual 'panic override' for the distro setup, but it doesn't replace the automation you could do (with the AAC) to reset such a system back to normal so I don't see any harm in it. I'm sure something creative could be to make pressure-sensitive, one-shot systems that don't rely on the AAC, which would be neat.

**Comment by ghost** [comment]
I like this. This will allow me to make time-delayed canister bombs a lot easier!

**Comment by Eneocho** [comment]
> I like this. This will allow me to make time-delayed canister bombs a lot easier!

duude, bbombs. t the bombs dude, dudebomb, the, dude hear me out, the bombs dude, dude, the bombs. dude bombs.

**Comment by Eneocho** [comment]
I like it. There is still a use for the AAC sensor+gate combo, but this helps make more compact atmos-based projects.

**Comment by adacovsk** [comment]
smart

**Comment by ShiftyRail** [comment]
https://www.youtube.com/watch?v=zyv16mZakVE

**Pull request merged**